mantidproject/mantid
# Indirect - CorrectionsTabs - Error handling in the case of a WorkspaceGroup being provided

~~**Must be merged after #21333**~~

Currently, in the Interfaces > Indirect > Corrections tabs, attempting to supply a WorkspaceGroup to the sample/container input fields, provides no informative error message. Upon clicking the 'Run' button in these interfaces, prior to supplying a WorkspaceGroup, Mantid will currently crash.

This PR adds correct error handling, in the circumstance in which a WorkspaceGroup, or any invalid workspace type is provided to the input fields of the Indirect CorrectionsTabs.

A message box will be displayed, when the invalid workspace type is first selected:
![corr_err_message](https://user-images.githubusercontent.com/15521164/33891018-b3adfcfa-df4c-11e7-982e-e94d15744288.PNG)

Along with a message output in the log:
![corr_err_log](https://user-images.githubusercontent.com/15521164/33891045-c91a6272-df4c-11e7-81a9-dc0fde511600.PNG)

This PR also improves the error message when the 'Run' button is clicked and a workspace of an invalid type, occupies the sample/container input field:
![corr_err_message_run](https://user-images.githubusercontent.com/15521164/33892838-c41e3276-df51-11e7-94a6-1a6412a37677.PNG)


**To test:**
1. Navigate to Interfaces > Indirect > Corrections.
2. For each tab in the Indirect Corrections interface:
    1. Supply the provided `sample_group_red.nxs` file into the sample field.
    2. Ensure an error message, similar to the one shown above, is displayed and a warning is output to the log.
    3. Click the 'Run' button.
    4. Ensure an error message, similar to the one shown above, is displayed, which includes a line about the sample input not being of the correct type.
    5. If there is a container field, supply the provided `container_group_red.nxs` file into the container field.
    6. In the 'ContainerSubtraction' interface, the same error message received when providing the sample group, should be produced. Otherwise, no error should be produced.
    7. Click the 'Run' button.
    8. Ensure an error message, similar to the one shown above, is displayed, which includes both a line about the sample input not being of the correct type and the same for the container input.

**Test Files:**
[group_test_files.zip](https://github.com/mantidproject/mantid/files/1552406/group_test_files.zip)



Fixes #21157. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
